Source/WebCore/svg/SVGTransformList.cpp

```diff
@@ -62,54 +62,71 @@ AffineTransform SVGTransformList::concatenate() const
     return result;
 }
 
-template<typename CharacterType> bool SVGTransformList::parseGeneric(StringParsingBuffer<CharacterType>& buffer)
+template<typename CharacterType> bool SVGTransformList::parseGeneric(StringParsingBuffer<CharacterType>& buffer, ListReplacement listReplacement)
 {
     bool delimParsed = false;
     skipOptionalSVGSpaces(buffer);
 
+    size_t itemIndex = 0;
+    auto currentListReplacement = listReplacement;
+
     while (buffer.hasCharactersRemaining()) {
         delimParsed = false;
-        
+
         auto parsedTransformType = SVGTransformable::parseTransformType(buffer);
         if (!parsedTransformType)
             return false;
 
-        auto parsedTransformValue = SVGTransformable::parseTransformValue(*parsedTransformType, buffer);
-        if (!parsedTransformValue)
+        RefPtr<SVGTransform> existingTransform;
+        if (currentListReplacement == ListReplacement::Replace && itemIndex < m_items.size() && parsedTransformType == m_items[itemIndex]->type())
+            existingTransform = m_items[itemIndex].ptr();
+
+        RefPtr parsedTransform = SVGTransformable::parseTransform(*parsedTransformType, buffer, existingTransform);
+        if (!parsedTransform)
             return false;
 
-        append(SVGTransform::create(WTFMove(*parsedTransformValue)));
+        if (currentListReplacement == ListReplacement::Replace && parsedTransform.get() != existingTransform.get()) {
+            currentListReplacement = ListReplacement::Append;
+            resize(itemIndex);
+        }
+
+        if (currentListReplacement == ListReplacement::Append)
+            append(WTFMove(parsedTransform.releaseNonNull()));
 
         skipOptionalSVGSpaces(buffer);
 
         if (skipExactly(buffer, ','))
             delimParsed = true;
 
         skipOptionalSVGSpaces(buffer);
+
+        ++itemIndex;
     }
+
+    if (currentListReplacement == ListReplacement::Replace && itemIndex < m_items.size())
+        resize(itemIndex);
```

Comment:
Good that you caught this case. But `itemIndex < m_items.size()` implies `currentListReplacement == ListReplacement::Replace` i.e. it is not possible that the first clause is true and the second one is false.